decidim-forms/app/commands/decidim/forms/admin/update_questionnaire.rb

```diff
@@ -78,9 +78,10 @@ def update_questionnaire_question(form_question)
               update_nested_model(form_display_condition, display_condition_attributes, question.display_conditions)
             end
 
-            form_question.matrix_rows.each do |form_matrix_row|
+            form_question.matrix_rows_by_position.each_with_index do |form_matrix_row, idx|
               matrix_row_attributes = {
-                body: form_matrix_row.body
+                body: form_matrix_row.body,
+                position: form_matrix_row.position || idx
```

Comment:
💬 I see a potential inconsistency here. In the migration you are using the `id` field to fill the position, but here if the position is null it's being filled with the index. Could it happen that a row without a position could be placed randomly depending on the ids of the other rows?
I feel that allowing `nil` values for the positions could generate more issues than fixing them. Wouldn't be simpler to keep the `position` as not `nil` (the migration should ensure that, right?) and use it to order the rows without `_with_index`? What do you think?

Reply:
@leio10  The migration fills the `position` column with ID because it will set a correct ordering. After the migration, when creating new questionnaires `position` will be an incremental value: 1, 2, ..

Here we have a single command that is used to `create` and to `update` matrix rows, that's why it has to consider if the  `position` attribute is `nil` or not.
The `position` field must keep the same value on each update, so it is not possible to have rows placed randomly. It is not allowed to have nil values on editing, it only happens when creating a new questionnaire.